### PR TITLE
New discussion form style

### PR DESCRIPTION
**Pull request by joaojusto**
screen shot 2015-01-08 at 17 28 29
screen shot 2015-01-08 at 17 28 54

### CODE REVIEW

File: app/assets/javascripts/app/discussions/discussions.module.js [hunk 1]
```diff
@@ -31,7 +31,28 @@
       })
       .state('discussions.new', {
         url: '/new',
-        templateUrl: 'discussions/new.html'
+        templateUrl: 'discussions/templates/discussion.html',
+        controller: 'NewDiscussionCtrl as ctrl'
+      })
+      .state('discussions.new.title', {
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/discussions.module.js]: I don't like these names, `title` and `body` don't seem like good names. What do you think about `step1` and `step2` for instance?

File: app/assets/javascripts/app/discussions/templates/edit_header_body.html.slim [hunk 2]
```diff
@@ -0,0 +1,6 @@
+md-button.md-primary.md-hue-1 ui-sref="discussions.new.title"
+  i.fa.fa-chevron-left.u-alpha
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/templates/edit_header_body.html.slim]: Why is there a `u-alpha` here? If you need a different type a button create a modifier.

File: app/assets/javascripts/app/discussions/templates/edit_header_title.html.slim [hunk 3]
```diff
@@ -0,0 +1,5 @@
+md-button.md-primary.md-hue-1 ui-sref="discussions.index"
+  i.fa.fa-times.u-alpha
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/templates/edit_header_title.html.slim]: Same thing here.

File: app/assets/javascripts/app/discussions/templates/header_body.html.slim [hunk 4]
```diff
@@ -0,0 +1,6 @@
+md-button.md-primary.md-hue-1 ui-sref="discussions.new.title"
+  i.fa.fa-chevron-left.u-alpha
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/templates/header_body.html.slim]: same here

File: app/assets/javascripts/app/discussions/templates/header_title.html.slim [hunk 5]
```diff
@@ -0,0 +1,5 @@
+md-button.md-primary.md-hue-1 ui-sref="discussions.index"
+  i.fa.fa-times.u-alpha
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/templates/header_title.html.slim]: the fact that you've done this 4 times must say something.

File: app/assets/javascripts/app/discussions/discussions.module.js (hunk 1 above)
Comment (joaojusto) [inline app/assets/javascripts/app/discussions/discussions.module.js]: That sounds better

File: app/assets/javascripts/app/discussions/templates/edit_header_body.html.slim (hunk 2 above)
Comment (joaojusto) [inline app/assets/javascripts/app/discussions/templates/edit_header_body.html.slim]: Its not a different type of button, its just font-size

File: app/assets/javascripts/app/discussions/templates/edit_header_title.html.slim (hunk 3 above)
Comment (joaojusto) [inline app/assets/javascripts/app/discussions/templates/edit_header_title.html.slim]: Same response as above

File: app/assets/javascripts/app/discussions/templates/edit_header_body.html.slim (hunk 2 above)
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/templates/edit_header_body.html.slim]: Exactly, it's a different button, and it's used 4 times for now. The utilities are two avoid and for things that are not repeated anywhere and you're sure will not.

Comment (joaojusto) [inline app/assets/javascripts/app/discussions/templates/edit_header_body.html.slim]: Ok I guess I can create a button

**Comment by joaojusto** [comment]
@gabrielpoca bump

File: app/assets/javascripts/app/discussions/edit.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    function updateDiscussion(discussion) {
+      Discussion.create(discussion).then(onDiscussionSave);
+    }
+
+    function onDiscussionSave(discussion) {
+      $state.go('discussions.show', { id: discussion.id });
+    }
+
+    $state.transitionTo('discussions.edit.step1', {
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/edit.js]: If you're using hoisted functions keep them on the bottom.

File: app/assets/javascripts/app/discussions/new.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    function createDiscussion(discussion) {
+      Discussion.create(discussion).then(onDiscussionSave);
+    }
+
+    function onDiscussionSave(discussion) {
+      $state.go('discussions.show', { id: discussion.id });
+    }
+
+    $state.transitionTo('discussions.new.step1');
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/new.js]: The same

File: app/assets/javascripts/app/discussions/templates/discussion.html.slim [hunk 8]
```diff
@@ -0,0 +1,8 @@
+.TopBarLayout
+  .TopBarLayout-header
+    md-toolbar.md-padding.md-whiteframe-z1.md-accent.u-gamma ui-view="header" layout="row" layout-align="space-between center"
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/templates/discussion.html.slim]: Why is `u-gamma` needed here? I don't remember it being necessary on the toolbar.

File: app/assets/javascripts/app/discussions/templates/edit_header_step1.html.slim [hunk 9]
```diff
@@ -0,0 +1,5 @@
+md-button.md-primary.md-hue-1 md-no-ink="" ui-sref="discussions.index"
+  span.fa.fa-times.u-alpha
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/templates/edit_header_step1.html.slim]: Why isn't this one an Icon?

File: app/assets/stylesheets/components/_icons.css.sass [hunk 10]
```diff
@@ -2,3 +2,11 @@
   font-family: FontAwesome
   font-size: 30px
   content: "\f0c9"
+
+.IconClose
+  display: inline-block
+
+  &:before
+    @include alpha
+    font-family: FontAwesome
```
Comment (gabrielpoca) [inline app/assets/stylesheets/components/_icons.css.sass]: You should probably refactor this into a mixin. Something that accepts the content and places the font-family also.

**Comment by gabrielpoca** [comment]
:+1:

**Pull request merged**